AsyncDisplayKit/ASControlNode.mm

```diff
@@ -424,5 +436,75 @@ - (void)cancelTrackingWithEvent:(UIEvent *)touchEvent
 - (void)endTrackingWithTouch:(UITouch *)touch withEvent:(UIEvent *)touchEvent
 {
 }
+#if TARGET_OS_TV
+#pragma mark - tvOS
+- (void)pressDown
+{
+  [UIView animateWithDuration:0.1 delay:0 options:UIViewAnimationCurveLinear animations:^{
+    [self setPressedState];
+  } completion:^(BOOL finished) {
+    if (finished) {
+      [UIView animateWithDuration:0.1 delay:0 options:UIViewAnimationCurveLinear animations:^{
+        [self setFocusedState];
+      } completion:nil];
+    }
+  }];
+}
+
+- (BOOL)canBecomeFocused
+{
+  return YES;
+}
+
+- (BOOL)shouldUpdateFocusInContext:(nonnull UIFocusUpdateContext *)context
+{
+  return YES;
+}
+
+- (void)didUpdateFocusInContext:(UIFocusUpdateContext *)context withAnimationCoordinator:(UIFocusAnimationCoordinator *)coordinator
+{
+  //FIXME: This is never valid inside an ASCellNode
+  if (context.nextFocusedView && context.nextFocusedView == self.view) {
+    //Focused
+    [coordinator addCoordinatedAnimations:^{
+      [self setFocusedState];
+    } completion:nil];
+  } else{
+    //Not focused
+    [coordinator addCoordinatedAnimations:^{
+      [self setDefaultState];
+    } completion:nil];
+  }
+}
+
+- (void)setFocusedState
+{
+  self.layer.shadowOffset = CGSizeMake(2, 10);
```

Comment:
Can you avoid repeated access to properties, at least all properties used 3 times or more in a method, by creating a local variable?